engine/gameobject/src/gameobject/comp_script.cpp

```diff
@@ -366,12 +366,28 @@ namespace dmGameObject
 
             int action_table = lua_gettop(L);
 
-            if (params.m_InputAction->m_IsGamepad) {
+            if (params.m_InputAction->m_IsGamepad) 
+            {
                 lua_pushliteral(L, "gamepad");
                 lua_pushnumber(L, params.m_InputAction->m_GamepadIndex);
                 lua_settable(L, action_table);
             }
 
+            if (params.m_InputAction->m_GamepadConnected) 
+            {
+                lua_pushliteral(L, "gamepad_name");
+                if (params.m_InputAction->m_TextCount == 0)
```

Comment:
No need for this if statement, since the textcount is 0, and should work well with the `lua_pushlstring` below.

Reply:
Fixed in https://github.com/defold/defold/pull/4970/commits/5a5b367269321c9773c1b9e96d219773ce07b9fe